### PR TITLE
Send company logo image for open position

### PR DESCRIPTION
**Pull request by LarisaBothaza**

### CODE REVIEW

File: src/main/java/com/sintern/api/CompanyController.java [hunk 1]
```diff
@@ -16,15 +20,23 @@
...
 
     @PostMapping("/register")
     @ResponseStatus(HttpStatus.OK)
     public void processRegister(@Valid @RequestBody CompanyRegisterRequest companyRegisterRequest) {
         companyService.addCompany(transform(companyRegisterRequest));
     }
+
+    @PostMapping("/upload-logo")
+    @ResponseStatus(HttpStatus.OK)
+    public void uploadCv(@RequestParam("id") UUID companyId, @RequestParam("image") MultipartFile image) {
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/CompanyController.java:39]: Are we uploading a CV here?

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 2]
```diff
@@ -1,37 +1,66 @@
...
 import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
 
 @CrossOrigin
 @RestController
 @RequestMapping("/open-positions")
 public class OpenInternPositionController {
-    private final OpenInternPositionService openInternPositionService;
+    private final OpenInternPositionServiceImpl openInternPositionService;
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: Why did you change this? We should use **interfaces**, not **concrete implementations**.

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 3]
```diff
@@ -1,37 +1,66 @@
...
 
     @RequestMapping(value = "/domains", method = RequestMethod.GET)
-    public List<Domain> getDomains(){
+    public List<Domain> getDomains() {
         return openInternPositionService.getDomains();
     }
+
+    @GetMapping("{openPositionId}/company-logo/")
+    public ResponseEntity<byte[]> getLogo(@PathVariable UUID openPositionId) {
+
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: Is this whitespace needed?

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 4]
```diff
@@ -1,37 +1,66 @@
...
         return openInternPositionService.findOpenInternPositionDTOByCompanyDomain(DomainType.valueOf(domainType));
     }
 
     @RequestMapping(value = "/domains", method = RequestMethod.GET)
-    public List<Domain> getDomains(){
+    public List<Domain> getDomains() {
         return openInternPositionService.getDomains();
     }
+
+    @GetMapping("{openPositionId}/company-logo/")
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: Does this work? The root doesn't end with "**/**" and this doesn't start with "**/**". I mean, the endpoint should be "**/open-positions/{openPositionId}/company-logo**", not "**/open-positions{openPositionId}/company-logo**". Not saying it's wrong, I'm just wondering if it works.

File: src/main/java/com/sintern/api/OpenInternPositionController.java [hunk 5]
```diff
@@ -1,37 +1,66 @@
...
+    public ResponseEntity<byte[]> getLogo(@PathVariable UUID openPositionId) {
+
+        Optional<CompanyLogo> companyLogoOptional = logoService.getLogoByCompanyId(openInternPositionService.getOpenInternPositionById(openPositionId).getCompany().getId());
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/api/OpenInternPositionController.java]: This is quite hard to read. Consider extracting `openInternPositionService.getOpenInternPositionById(openPositionId).getCompany()` to a variable.

File: src/main/java/com/sintern/service/LogoServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    private final CompanyRepository companyRepository;
+
+    @Autowired
+    public LogoServiceImpl(LogoRepository logoRepository, CompanyRepository companyRepository) {
+        this.logoRepository = logoRepository;
+        this.companyRepository = companyRepository;
+    }
+
+    @Override
+    @Transactional
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/service/LogoServiceImpl.java:31]: Just curious. Why is `@Transactional` needed here?

File: src/main/java/com/sintern/service/OpenInternPositionService.java [hunk 7]
```diff
@@ -1,58 +1,19 @@
...
-import org.springframework.stereotype.Service;
+import com.sintern.domain.enums.DomainType;
 
-import java.util.ArrayList;
 import java.util.List;
-import java.util.stream.Collectors;
-
-@Service
-public class OpenInternPositionService {
-    private final OpenInternPositionRepository openInternPositionRepository;
```
Comment (DascaluCosmin) [inline src/main/java/com/sintern/service/OpenInternPositionService.java:17]: Not saying this is wrong, but it's **out-of-scope** for this story. For such a refactor, we would have a separate **fix** branch. We try to keep only the relevant code for the story we're working on.

File: src/main/java/com/sintern/api/OpenInternPositionController.java (hunk 2 above)
Comment (LarisaBothaza) [reply src/main/java/com/sintern/api/OpenInternPositionController.java]: My inattention. There is no interface. I added it and refactored it automatically.

File: src/main/java/com/sintern/api/OpenInternPositionController.java (hunk 4 above)
Comment (LarisaBothaza) [reply src/main/java/com/sintern/api/OpenInternPositionController.java]: It works even if it doesn't start with "/". I delete the one at the end.

File: src/main/java/com/sintern/service/LogoServiceImpl.java (hunk 6 above)
Comment (LarisaBothaza) [reply src/main/java/com/sintern/service/LogoServiceImpl.java:31]: in short: _Optional_ _Company Logo_ is returned

File: src/main/java/com/sintern/service/OpenInternPositionService.java (hunk 7 above)
Comment (LarisaBothaza) [reply src/main/java/com/sintern/service/OpenInternPositionService.java:17]: For me, it's not out-of-scope for this story. I had to use the OpenInternaPositionService and I found that the interface was missing. But ok, in the future I will let you know when I see any more irregularities. Or change back?

Comment (DascaluCosmin) [reply src/main/java/com/sintern/service/OpenInternPositionService.java:17]: You leave the way it is and ping me.
